pkg/client/client.go

```diff
@@ -59,6 +59,7 @@ type Client struct {
 	Discovery              discovery.DiscoveryInterface
 	DeploymentCli          appsv1.AppsV1Interface
 	KubernetesExtensionCli kubernetes.Interface
+	HTTPCli                restclient2.HTTPClient
```

Comment:
I believe we should not add this attribute to the `Client` structure since it's just a `http.Client` instance dancing around the functions. It's safer to create it once we need it, plus it's not associated with any k8s client whatsoever.

Reply:
I had removed it from `Client` and directly call `http.Get` when required